tracim/tracim/lib/email_fetcher.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import time
+import imaplib
+import datetime
+import json
+import typing
+from email.message import Message
+from email.header import Header, decode_header, make_header
+from email.utils import parseaddr, parsedate_tz, mktime_tz
+from email import message_from_bytes
+
+import markdown
+import requests
+from bs4 import BeautifulSoup
+from email_reply_parser import EmailReplyParser
+
+from tracim.lib.base import logger
+
+TRACIM_SPECIAL_KEY_HEADER = "X-Tracim-Key"
+BS_HTML_BODY_PARSE_CONFIG = {
+    'tag_blacklist': ["script", "style", "blockquote"],
+    'class_blacklist': ['moz-cite-prefix', 'gmail_extra', 'gmail_quote',
+                        'yahoo_quoted'],
+    'id_blacklist': ['reply-intro'],
+    'tag_whitelist': ['a', 'b', 'strong', 'i', 'br', 'ul', 'li', 'ol',
+                      'em', 'i', 'u',
+                      'thead', 'tr', 'td', 'tbody', 'table', 'p', 'pre'],
+    'attrs_whitelist': ['href'],
+}
+
+
+class DecodedMail(object):
+
+    def __init__(self, message: Message):
+        self._message = message
+
+    def _decode_header(self, header_title: str) -> typing.Optional[str]:
+        # FIXME : Handle exception
+        if header_title in self._message:
+            return str(make_header(decode_header(header)))
+        else:
+            return None
+
+    def get_subject(self) -> typing.Optional[str]:
+        return self._decode_header('subject')
+
+    def get_from_address(self) -> typing.Optional[str]:
+        return parseaddr(self._message['From'])[1]
+
+    def get_to_address(self)-> typing.Optional[str]:
+        return parseaddr(self._message['To'])[1]
+
+    def get_first_ref(self) -> typing.Optional[str]:
+        return parseaddr(self._message['References'])[1]
+
+    def get_special_key(self) -> typing.Optional[str]:
+        return self._decode_header(TRACIM_SPECIAL_KEY_HEADER)
+
+    def get_body(self) -> typing.Optional[str]:
+        body_part = self._get_mime_body_message()
+        body = None
+        if body_part:
+            charset = body_part.get_content_charset('iso-8859-1')
+            ctype = body_part.get_content_type()
+            if ctype == "text/plain":
+                txt_body = body_part.get_payload(decode=True).decode(
+                    charset)
+                body = DecodedMail._parse_txt_body(txt_body)
+
+            elif ctype == "text/html":
+                html_body = body_part.get_payload(decode=True).decode(
+                    charset)
+                body = DecodedMail._parse_html_body(html_body)
+
+        return body
+
+    @staticmethod
+    def _parse_txt_body(txt_body: str):
+        txt_body = EmailReplyParser.parse_reply(txt_body)
+        html_body = markdown.markdown(txt_body)
```

Comment:
You convert txt content into html content to be able to use your function `_parse_html_body` or for another reason ? If yes a comment should be great : )